patchwork-events-world/src/main/java/com/patchworkmc/mixin/event/world/MixinMinecraftServer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.mixin.event.world;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.WorldEvent;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.ServerTask;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.NonBlockingThreadExecutor;
+import net.minecraft.world.dimension.DimensionType;
+
+@Mixin(MinecraftServer.class)
+public abstract class MixinMinecraftServer extends NonBlockingThreadExecutor<ServerTask> {
+	public MixinMinecraftServer(String name) {
+		super(name);
+	}
+
+	@Shadow
+	@Final
+	private Map<DimensionType, ServerWorld> worlds;
+
+	// Should get called once per loop, regardless of which if branch it takes.
+	//	@Inject(
+	//		method = "createWorlds",
+	//		slice = @Slice(
+	//			from = @At(value = "INVOKE", target = "java/util/Iterator.hasNext ()Z")
+	//		),
+	//		at = @At(value = "JUMP", opcode = Opcodes.GOTO),
+	//		locals = LocalCapture.CAPTURE_FAILHARD
+	//	)
+	//	private void hookCreateWorlds(WorldSaveHandler worldSaveHandler, LevelProperties properties, LevelInfo levelInfo, WorldGenerationProgressListener worldGenerationProgressListener, CallbackInfo ci, ServerWorld serverWorld, ServerWorld serverWorld2, Iterator var7, DimensionType dimensionType) {
+	//		MinecraftForge.EVENT_BUS.post(new WorldEvent.Load(this.worlds.get(dimensionType)));
+	//	}
+
+	// Alternatively, mixin to the put call, and special case overworld.
+	// Perhaps move the special case outside of the loop?
+	@Redirect(method = "createWorlds", at = @At(value = "INVOKE", target = "java/util/Iterator.next ()Ljava/lang/Object;"))
```

Comment:
So, I ended up creating two separate possible mixins for this. I currently like the latter more, as I feel like it's more robust, but I'm leaving the former in for now, in case its actually better. I will remove whichever one isn't used before merging.